engine-types/src/parameters/engine.rs

```diff
@@ -575,51 +570,47 @@ mod tests {
     }
 
     #[test]
-    fn test_serialization_order_transaction_status() {
-        let res = borsh::to_vec(&TransactionStatus::Succeed(Vec::new())).unwrap();
-        assert_eq!(&[0, 0, 0, 0, 0], res.as_slice());
-
-        let res = borsh::to_vec(&TransactionStatus::Revert(Vec::new())).unwrap();
-        assert_eq!(&[1, 0, 0, 0, 0], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::OutOfGas).unwrap();
-        assert_eq!(&[2], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::OutOfFund).unwrap();
-        assert_eq!(&[3], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::OutOfOffset).unwrap();
-        assert_eq!(&[4], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::CallTooDeep).unwrap();
-        assert_eq!(&[5], res.as_slice());
-
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::StackUnderflow)).unwrap();
-        assert_eq!(&[6, 0], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::StackOverflow)).unwrap();
-        assert_eq!(&[6, 1], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::InvalidJump)).unwrap();
-        assert_eq!(&[6, 2], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::InvalidRange)).unwrap();
-        assert_eq!(&[6, 3], res.as_slice());
-        let res =
-            borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::DesignatedInvalid)).unwrap();
-        assert_eq!(&[6, 4], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::CreateCollision)).unwrap();
-        assert_eq!(&[6, 5], res.as_slice());
-        let res =
-            borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::CreateContractLimit)).unwrap();
-        assert_eq!(&[6, 6], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::InvalidCode(0))).unwrap();
-        assert_eq!(&[6, 7, 0], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::PCUnderflow)).unwrap();
-        assert_eq!(&[6, 8], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::CreateEmpty)).unwrap();
-        assert_eq!(&[6, 9], res.as_slice());
-        let res = borsh::to_vec(&TransactionStatus::Error(EvmErrorKind::Other(
-            crate::Cow::from(String::new()),
-        )))
+    fn test_serialization_transaction_status_regression() {
+        let bytes =
+            std::fs::read("../engine-tests/src/tests/res/transaction_status.borsh").unwrap();
```

Comment:
I understand, why that solution was proposed: some guaranty or reference of transaction order.
But nothing can prevent the change that file. It means that the serialization generator can be used directly, as it will be difficult to miss ordering modifications for it.
Also keep in mind, that new EVM errors will be frequently adding.